### PR TITLE
samples: matter: Enabled using Thread library configs

### PR DESCRIPTION
**Pull request by kkasperczyk-no**
* Removed custom OpenThread configuration from samples and enabled OpenThread Nordic library MTD/FTD configs
* Changed light_bulb default config from MTD to FTD

KRKNWK-12776
KRKNWK-12954

Signed-off-by: Kamil Kasperczyk <kamil.kasperczyk@nordicsemi.no>

### CODE REVIEW

File: samples/matter/light_bulb/src/app_task.cpp [hunk 1]
```diff
@@ -103,7 +103,15 @@ CHIP_ERROR AppTask::Init()
 		return err;
 	}
 
+#ifdef CONFIG_OPENTHREAD_MTD
+#ifdef CONFIG_OPENTHREAD_MTD_SED
```
Comment (Damian-Nordic) [inline samples/matter/light_bulb/src/app_task.cpp:107]: `OPENTHREAD_MTD_SED` depends on `OPENTHREAD_MTD` so I think we can reduce nesting of `#ifdefs` here.

File: samples/matter/light_switch/prj.conf [hunk 2]
```diff
@@ -19,12 +19,7 @@ CONFIG_NET_SOCKETS_POLL_MAX=5
 # Enable OpenThread MTD
 CONFIG_NET_L2_OPENTHREAD=y
 
-CONFIG_OPENTHREAD_FTD=y
-CONFIG_OPENTHREAD_SLAAC=y
-CONFIG_OPENTHREAD_MTD_NETDIAG=y
-CONFIG_OPENTHREAD_MANUAL_START=y
-CONFIG_OPENTHREAD_THREAD_STACK_SIZE=6144
-CONFIG_OPENTHREAD_NUM_MESSAGE_BUFFERS=64
+CONFIG_OPENTHREAD_NORDIC_LIBRARY_FTD=y
```
Comment (Damian-Nordic) [inline samples/matter/light_switch/prj.conf:22]: Should the light switch be MTD?

File: samples/matter/light_bulb/src/app_task.cpp (hunk 1 above)
Comment (kkasperczyk-no) [reply samples/matter/light_bulb/src/app_task.cpp:107]: Yes, I know it does, but how would we separate sleepy MTD from non-sleepy MTD in that case?

Comment (kkasperczyk-no) [reply samples/matter/light_bulb/src/app_task.cpp:107]: Ok, light bulb doesn't even support SED configuration, so i will just remove it

Comment (Damian-Nordic) [reply samples/matter/light_bulb/src/app_task.cpp:107]: I meant sth like:
```
#ifdef CONFIG_OPENTHREAD_MTD_SED

#elif CONFIG_OPENTHREAD_MTD

#else
```

**Review by Damian-Nordic — APPROVED** [review]

**Review by Damian-Nordic — APPROVED** [review]

**Comment by kkasperczyk-no** [comment]
Rebased

**Review by rlubos — APPROVED** [review]

**Pull request merged**